docs/docs/persistent-volumes.md

```diff
@@ -1,86 +1,303 @@
+
+Once a task that used persistent volumes has terminated, its metadata will be kept. This metadata will be used to launch a replacement task when needed.
+
+For example, if you scale down from 5 to 3 instances, you will see 2 tasks in the `Waiting` state along with the information about the persistent volumes the tasks were using as well as about the agents on which they are placed. Marathon will not unreserve those resources and will not destroy the volumes. When you scale up again, Marathon will attempt to launch tasks that use those existing reservations and volumes as soon as it gets a Mesos offer containing the labeled resources. Marathon will only schedule unreserve/destroy operations when:
+
+- the application is deleted (in which case volumes of all its tasks are destroyed, and all reservations are deleted).
+- you explicitly delete one or more suspended tasks with a `wipe=true` flag.
+
+If reserving resources or creating persistent volumes fails, the created task will timeout after the configured `task_reservation_timeout` (default: 20 seconds) and a new reservation attempt will be made. In case a task is `LOST` (because its agent is disconnected or crashed), the reservations and volumes will not timeout and you need to manually delete and wipe the task in order to let Marathon launch a new one.
+
+## Potential Pitfalls
+
+Be aware of the following issues and limitations when using stateful applications in Marathon that make use of dynamic resevations and persistent volumes.
+
+### Static Reservations
+
+Dynamic reservations can only be created for unreserved resources. If you specify an agent's resources to be reserved for a role via the Mesos `--resources` or `--default_role` flag, these resources cannot be used for dynamic reservations. In addition, if Marathon is started with the `--default_accepted_resource_roles` flag specifying a value that does not contain `*`, your application definition should explicitly specify `"acceptedResourceRoles": ["*"]` in order to allow usage and reservation of unreserved cluster resources.
+
+### Resource requirements
+
+Currently, the resource requirements of a stateful application **cannot** be changed. Your initial volume size, cpu usage, memory requirements, etc., cannot be changed once you've posted the AppDefinition.
+
+### Replication and Backups
+
+Because persistent volumes are pinned to nodes, they are no longer reachable if the node is disconnected from the cluster, e.g. due to a network partition or a crashed agent. If the stateful service does not take care of data replication on its own, you need to manually setup a replication or backup strategy to guard against data loss from a network partition or from a crashed agent.
```

Comment:
I tried to clarify this a little. It's really up to the user to setup _some_ data replication or backup strategy, and we don't provide any help for achieving that.

A task will eventually be `LOST` during a network partition, but that's not yet data loss. Even a crashed agent doesn't mean data loss if the agent can be restarted and the metadata about the volumes is still there and will be offered again. However, the service will not be able to use the data while the agent is down or unreachable, so it might be out of date when the agent is reconnected.